aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/helpers/AuthHelper.kt

```diff
@@ -55,5 +55,16 @@ internal open class AuthHelper {
                     }
             }
         }
+
+        /**
+         * This method is used to capture the activeUsername we should be using to store and retreive device Metadata.
+         * This becomes more important when an alias is used to signin instead of a username where cognito then
+         * generates the username on the customers behalf and returns it. We need to then use that username to store
+         * and retrieve device information as the username that the customer entered will no longer be available when
+         * respondToAuthChallenge/confirmSignIn calls are made.
+         * */
+        fun getActiveUsername(username: String, alternateUsername: String?, userIDForSRP: String?): String {
```

Comment:
May make sense to use the map of challenge parameters here as a map so that we aren't parsing out the usernames in multiple places

Reply:
Instead of passing the map around I thought might be better to just pass in the strings. The parsing has to be done either way in all the places. Are you suggesting for better readability or are there performance concerns too ?

Reply:
Discussed offline and agreed upon to use primitives instead of map